provider/vsphere/internal/vsphereclient/createvm_test.go

```diff
@@ -359,15 +346,28 @@ func (s *clientSuite) TestCreateVirtualMachineExternalNetworkSpecifiedDVPortgrou
 	})
 }
 
-func (s *clientSuite) TestCreateVirtualMachineExternalNetworkNotFound(c *gc.C) {
+func (s *clientSuite) TestCreateVirtualMachineNetworkNotFound(c *gc.C) {
 	args := baseCreateVirtualMachineParams(c)
-	args.ExternalNetwork = "fourtytwo"
+	args.NetworkDevices = []NetworkDevice{
+		NetworkDevice{Network: "fourtytwo"},
+	}
 
 	client := s.newFakeClient(&s.roundTripper, "dc0")
 	_, err := client.CreateVirtualMachine(context.Background(), args)
 	c.Assert(err, gc.ErrorMatches, `creating import spec: network "fourtytwo" not found`)
 }
 
+func (s *clientSuite) TestCreateVirtualMachineInvalidMAC(c *gc.C) {
+	args := baseCreateVirtualMachineParams(c)
+	args.NetworkDevices = []NetworkDevice{
+		NetworkDevice{MAC: "00:11:22:33:44:55"},
+	}
+
+	client := s.newFakeClient(&s.roundTripper, "dc0")
+	_, err := client.CreateVirtualMachine(context.Background(), args)
+	c.Assert(err, gc.ErrorMatches, `creating import spec: adding network device 0 - network VM Network: Invalid MAC address: "00:11:22:33:44:55"`)
```

Comment:
for anyone seeing this, would it be useful to explain what would make a valid MAC address.